BlueShift-iOS-SDK/BlueShift.m

```diff
@@ -42,113 +58,122 @@ - (void)setAppDelegate {
-        } else {
-            blueShiftAppDelegate.userNotificationDelegate = blueShiftUserNotificationCenterDelegate;
+    @try {
+        // Validate the API key
+        if ([config validateConfigDetails] == NO) {
+            return ;
         }
-    }
-    
-    if(config.blueShiftPushDelegate) {
-        blueShiftAppDelegate.blueShiftPushDelegate = config.blueShiftPushDelegate;
-    }
-    if (config.blueshiftUniversalLinksDelegate) {
-        blueShiftAppDelegate.blueshiftUniversalLinksDelegate = config.blueshiftUniversalLinksDelegate;
-    }
-    if (config.blueshiftDeviceIdSource) {
-        [[BlueShiftDeviceData currentDeviceData] setBlueshiftDeviceIdSource:config.blueshiftDeviceIdSource];
         
-        //Custom device id provision for DeviceIDSourceCUSTOM
-        if (config.blueshiftDeviceIdSource == BlueshiftDeviceIdSourceCustom) {
-            if (config.customDeviceId && ![config.customDeviceId isEqualToString:@""]) {
-                [[BlueShiftDeviceData currentDeviceData] setCustomDeviceID:config.customDeviceId];
-                [BlueshiftLog logInfo: [NSString stringWithFormat:@"CUSTOM device id is set as - %@",config.customDeviceId] withDetails:nil methodName:nil];
+        // Set config
+        _sharedBlueShiftInstance.config = config;
+        
+        // Set up device id
+        if (config.blueshiftDeviceIdSource) {
+            [[BlueShiftDeviceData currentDeviceData] setBlueshiftDeviceIdSource:config.blueshiftDeviceIdSource];
+            
+            //Custom device id provision for DeviceIDSourceCUSTOM
+            if (config.blueshiftDeviceIdSource == BlueshiftDeviceIdSourceCustom) {
+                if (config.customDeviceId && ![config.customDeviceId isEqualToString:@""]) {
+                    [[BlueShiftDeviceData currentDeviceData] setCustomDeviceID:config.customDeviceId];
+                    [BlueshiftLog logInfo: [NSString stringWithFormat:@"CUSTOM device id is set as - %@",config.customDeviceId] withDetails:nil methodName:nil];
+                } else {
+                    [BlueshiftLog logError:nil withDescription:@"ERROR: CUSTOM device id is not provided" methodName:nil];
+                }
+            }
+        }
+        
+        // Set BlueshiftAppDelegate
+        _sharedBlueShiftInstance.appDelegate = [[BlueShiftAppDelegate alloc] init];
+        _sharedBlueShiftInstance.appDelegate.mainAppDelegate = [UIApplication sharedApplication].delegate;
+        
+        // Initialise core data
+        [_sharedBlueShiftInstance.appDelegate initializeCoreData];
+        
+        // Set up Push notifications and delegates
+        BlueShiftUserNotificationCenterDelegate *blueShiftUserNotificationCenterDelegate = [[BlueShiftUserNotificationCenterDelegate alloc] init];
+        _sharedBlueShiftInstance.userNotificationDelegate = blueShiftUserNotificationCenterDelegate;
+        
+        if (@available(iOS 8.0, *)) {
```

Comment:
our new target version is now 9+ right? does this line of code will ever get executed?

Reply:
Yes, I had it added before making the pod changes. Will fix it.